tests/berlin/eip2929_gas_cost_increases/test_precompile_warming.py

```diff
@@ -0,0 +1,147 @@
+"""
+abstract: Tests [EIP-2929: Gas cost increases for state access opcodes](https://eips.ethereum.org/EIPS/eip-2929)
+    Test cases for [EIP-2929: Gas cost increases for state access opcodes](https://eips.ethereum.org/EIPS/eip-2929).
+"""  # noqa: E501
+
+from typing import Iterator, Tuple
+
+import pytest
+
+from ethereum_test_forks import (
+    Fork,
+    get_transition_fork_predecessor,
+    get_transition_fork_successor,
+)
+from ethereum_test_tools import (
+    Account,
+    Address,
+    Alloc,
+    Block,
+    BlockchainTestFiller,
+    Transaction,
+)
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+REFERENCE_SPEC_GIT_PATH = "EIPS/eip-2929.md"
+REFERENCE_SPEC_VERSION = "0e11417265a623adb680c527b15d0cb6701b870b"
+
+
+def precompile_addresses_in_predecessor_successor(
+    fork: Fork,
+) -> Iterator[Tuple[Address, bool, bool]]:
+    """
+    Yield the addresses of precompiled contracts and whether they existed in the parent fork.
+
+    Args:
+        fork (Fork): The transition fork instance containing precompiled contract information.
+
+    Yields:
+        Iterator[Tuple[str, bool]]: A tuple containing the address in hexadecimal format and a
+            boolean indicating whether the address has existed in the predecessor.
+
+    """
+    predecessor_precompiles = set(get_transition_fork_predecessor(fork).precompiles())
+    successor_precompiles = set(get_transition_fork_successor(fork).precompiles())
+    all_precompiles = successor_precompiles | predecessor_precompiles
+    highest_precompile = int.from_bytes(max(all_precompiles))
+    extra_range = 32
+    extra_precompiles = {
+        Address(i) for i in range(highest_precompile + 1, highest_precompile + extra_range)
+    }
+    all_precompiles = all_precompiles | extra_precompiles
+    for address in sorted(all_precompiles):
+        yield address, address in successor_precompiles, address in predecessor_precompiles
+
+
+@pytest.mark.valid_at_transition_to("Paris", subsequent_forks=True)
```

Comment:
Shouldn't this be Berlin? These tests should start to produce meaningful results once EIP 2929 is introduced. (One could also test before that of course, for forks introducing new precompiles but without 2929 active the gas difference would not depend upon warm/cold precompiles, but rather re-pricing of the opcodes used in the test (which are thus not subject of what we want to test here))

Reply:
Ah ok, before 2929 (Berlin) precompiles got added in Istanbul and Byzantium but those are thus not subject of the warm/cold tests. So marking it as "Paris" is fine here. Since Berlin, only the KZG (Cancun) and BLS (Prague) were introduced